### PR TITLE
Save score change

### PR DESCRIPTION
**Pull request by SClugston9**

### CODE REVIEW

File: src/controller/GameController.py [hunk 1]
```diff
@@ -393,4 +414,9 @@ def processGuess(self, userGuess: str):
         self.myUserInterface.showCorrectGuess()
         if currentPoints == maxPoints:
             self.myUserInterface.showEnd()
+            confirm = self.myUserInterface.getConfirmation("You are in the top 10 players for this puzzle! Do you want to save your score?")
+            if confirm == "y" or confirm == "Yes":
+                self.myDataSource.setHighScore(self.myPuzzle.getPuzzleLetters(), self.myUserInterface.getScoreName(), self.myPuzzle.getCurrentPoints())
+                self.myPuzzle.setHighScores(self.myDataSource.getHighScores(self.myPuzzle.getPuzzleLetters()))
+                self.myUserInterface.showMessage("Congrats! your score is now entered into the top 10 leaderboard for this puzzle!\n")
```
Comment (b0balagurak) [inline src/controller/GameController.py:406]: caps my man

File: src/controller/GameController.py [hunk 2]
```diff
@@ -184,6 +195,16 @@ def processCommand(self, command: Commands) -> None:
 
         elif command == Commands.EXIT:
             if self.playing:
+                if self.myPuzzle.getCurrentPoints() > self.myPuzzle.getMinimumHighScore(): 
+                    confirm = self.myUserInterface.getConfirmation("You are in the top 10 players for this puzzle! Do you want to save your score?\nWARNING- saving your score you will no longer be able to play this puzzle")
+                    if confirm == "y" or confirm == "Yes":
+                        self.myDataSource.setHighScore(self.myPuzzle.getPuzzleLetters(), self.myUserInterface.getScoreName(), self.myPuzzle.getCurrentPoints())
+                        self.myPuzzle.setHighScores(self.myDataSource.getHighScores(self.myPuzzle.getPuzzleLetters()))
+                        self.myUserInterface.showMessage("Congrats! your score is now entered into the top 10 leaderboard for this puzzle!\n")
```
Comment (b0balagurak) [inline src/controller/GameController.py]: here too

File: src/controller/GameController.py [hunk 3]
```diff
@@ -174,6 +174,17 @@ def processCommand(self, command: Commands) -> None:
 
         if command == Commands.QUIT:
             if self.playing:
+                if self.myPuzzle.getCurrentPoints() > self.myPuzzle.getMinimumHighScore(): 
+                    confirm = self.myUserInterface.getConfirmation("You are in the top 10 players for this puzzle! Do you want to save your score?\nWARNING- saving your score you will no longer be able to play this puzzle")
+                    if confirm == "y" or confirm == "Yes":
+                        self.myDataSource.setHighScore(self.myPuzzle.getPuzzleLetters(), self.myUserInterface.getScoreName(), self.myPuzzle.getCurrentPoints())
+                        self.myPuzzle.setHighScores(self.myDataSource.getHighScores(self.myPuzzle.getPuzzleLetters()))
+                        self.myUserInterface.showMessage("Congrats! your score is now entered into the top 10 leaderboard for this puzzle!\n")
```
Comment (b0balagurak) [inline src/controller/GameController.py]: bruh

**Review by b0balagurak — APPROVED** [review]
aight

**Review by nrhoopes — APPROVED** [review]

**Pull request merged**